packages/kbn-es-archiver/src/actions/empty_kibana_index.ts

```diff
@@ -8,25 +8,15 @@
 
 import type { Client } from '@elastic/elasticsearch';
 import { ToolingLog } from '@kbn/tooling-log';
-import { KbnClient } from '@kbn/test';
 
 import { ALL_SAVED_OBJECT_INDICES } from '@kbn/core-saved-objects-server';
-import { migrateSavedObjectIndices, createStats, cleanSavedObjectIndices } from '../lib';
+import { createStats, cleanSavedObjectIndices } from '../lib';
 
-export async function emptyKibanaIndexAction({
-  client,
-  log,
-  kbnClient,
-}: {
-  client: Client;
-  log: ToolingLog;
-  kbnClient: KbnClient;
-}) {
+export async function emptyKibanaIndexAction({ client, log }: { client: Client; log: ToolingLog }) {
   const stats = createStats('emptyKibanaIndex', log);
 
   await cleanSavedObjectIndices({ client, stats, log });
-  await migrateSavedObjectIndices(kbnClient);
   await client.indices.refresh({ index: ALL_SAVED_OBJECT_INDICES });
-  ALL_SAVED_OBJECT_INDICES.forEach((indexPattern) => stats.createdIndex(indexPattern));
+
   return stats.toJSON();
```

Comment:
At the moment we call SO migration during esArchive loading and only if it modified `.kibana` index
https://github.com/elastic/kibana/blob/a29e4aa38b29dfbe03e3cc00df47e8c7aa34a0a7/packages/kbn-es-archiver/src/actions/load.ts#L109-L118

If I understand correctly, in `.kibana` cleanup SO migration was from the start (added 5 years ago):
https://github.com/elastic/kibana/commit/c64624ada3dae195da688178d4d9433ee3e49049#diff-b22ea1611642a7e2cdc07bce3086e1b7e260f075fb08fdc9c2ae67ecf3742444R23-R24 

>Performing a migration as part of the cleanup does not make too much sense (at least in this test suite)

Could you provide more context here. Since many tests call `await esArchiver.emptyKibanaIndex();` I would like to understand if not running SO migrations is desired behaviour for cleanup or it should be optionally run for some cases?

Reply:
Many of those `esArchiver.emptyKibanaIndex()` calls were added in https://github.com/elastic/kibana/pull/85778

```
Enables v2 migrations in all tests
Because of removing the auto-create index behaviour I had to adapt tests to always initialize an empty kibana with esArchiver.emptyKibanaIndex() or move some before statements around so that we don't set UiSettings before the index gets initialized.
```

@rudolf do you think we can safely remove migration from that call and be sure it will work for all those cases?

Reply:
Let's consider the following groups of tests that use `esArchiver`:

- **1. Tests where we have deleted the `mappings.json` definitions.** We are implicitly using the "official" saved object indices created by Kibana at startup, which are already aligned with the current stack version. In these cases, there's no need to "realign" them. On the contrary, doing so triggers unnecessary migrations that can cause errors + flakiness.

- **2. Tests that don't define saved object indices.** There are a few tests that use `esArchiver` archives that DON'T define saved object indices (they define other indices). For these tests, we fall in the same scenario as (1), i.e. we are not recreating the "official" indices created by Kibana, and thus there's no need to "realign" them.

- **3. Tests that define `.kibana` index (and perhaps other saved object indices) in `mappings.json`. If some of these tests are calling `esArchiver.emptyKibanaIndex()`, they could be impacted by this PR, as we are no longer migrating on cleanup. But so far CI hasn't complained, and I'm on [a crusade](https://github.com/elastic/kibana/issues/161882) to remove them all.

Reply:
yeah quite a lot has changed since https://github.com/elastic/kibana/pull/85778 and I agree with @gsoldevila that (3) should be the only tests likely to be impacted.

Reply:
Thanks for the details, especially great to hear you are working on those tests removal.

@gsoldevila  I guess there is not much we can do in esArchiver code to prevent someone from adding new test suites with loading SO indexes and calling `emptyKibanaIndex` later in the test? We still need to load SO indexes with custom mappings for some scenarios, right?
Maybe leaving a comment to the `emptyKibanaIndex` function is a good thing to have.

Reply:
Actually, if/when I manage to remove all SO index definitions from mappings.json, I can throw an exception from `esArchiver` side, saying such indices cannot be defined.